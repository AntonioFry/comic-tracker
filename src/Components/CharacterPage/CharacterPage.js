import React, { Component } from 'react';

class CharacterPage extends Component {
  constructor() {
    super()
    this.state = {
      searchValue: '',
    }
  }

  changeSearchValue = (search) => {
    this.setState({ searchValue: search });
  }

  render() {
    return (
      <section className="character-section">
        
      </section>
    )
  }
}

export default CharacterPage;