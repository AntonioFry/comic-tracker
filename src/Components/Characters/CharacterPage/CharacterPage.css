.character-card-section {
  height: 500px;
}