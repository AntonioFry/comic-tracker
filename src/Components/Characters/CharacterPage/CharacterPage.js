import React, { Component } from 'react';
import { SearchBar } from '../../SearchBar/SearchBar';
import { getCharacters } from '../../../API/apicalls';
import './CharacterPage.css'
import { CharacterCard } from '../CharacterCard/CharacterCard';

class CharacterPage extends Component {
  constructor() {
    super()
    this.state = {
      characters: [],
    }
  }

  searchCharacter = async (name) => {
    const characters = await getCharacters(name);
    await this.setState({ characters });
    console.log(this.state.characters)
  }

  render() {
    const characterCards = this.state.characters.map(character => {
      return (
        <CharacterCard
          name={character.name}
          thumbnail={character.thumbnail}
          id={character.id}
        />
      )
    });
    return (
      <section className="character-section">
        <SearchBar searchCharacter={this.searchCharacter} />
        <section className='character-card-section'>
          { this.state.characters !== [] ? characterCards : null } 
        </section>
      </section>
    )
  }
}

export default CharacterPage;