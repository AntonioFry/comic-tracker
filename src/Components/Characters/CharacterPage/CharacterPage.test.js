import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import { CharacterPage } from './CharacterPage.js';
import { BrowserRouter } from 'react-router-dom';

describe('CharacterPage', () => {

  it('Should match the snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <CharacterPage />
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

})