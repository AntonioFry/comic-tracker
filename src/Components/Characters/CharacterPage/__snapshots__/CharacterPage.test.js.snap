exports[`CharacterPage Should match the snapshot 1`] = `
<section
  className="character-section">
  <form
    className="search-form">
    <input
      className="search-input"
      name="inputValue"
      onChange={[Function]}
      placeholder="Character Name..."
      value="" />
    <button
      className="search-button"
      onClick={[Function]}>
      Search
    </button>
  </form>
  <section
    className="character-card-section" />
</section>
`;
